homeassistant/components/ovo_energy/config_flow.py

```diff
@@ -111,19 +115,13 @@ async def async_step_reauth(
                 errors["base"] = "connection_error"
             else:
                 if authenticated:
-                    entry = await self.async_set_unique_id(self.username)
-                    if entry:
-                        self.hass.config_entries.async_update_entry(
-                            entry,
-                            data={
-                                CONF_USERNAME: self.username,
-                                CONF_PASSWORD: user_input[CONF_PASSWORD],
-                            },
-                        )
-                        return self.async_abort(reason="reauth_successful")
+                    return self.async_update_reload_and_abort(
+                        self._get_reauth_entry(),
+                        data_updates={CONF_PASSWORD: user_input[CONF_PASSWORD]},
```

Comment:
Using data_updates to update just the password.
As a side-effect, this also means that the custom account doesn't get lost during the reauth process - I'm pretty sure that was a bug.